CRM/PivotReport/Upgrader.php

```diff
@@ -295,22 +295,22 @@ private function addPermissionForMenuItem($menuItemName, $newPermission, $permis
    * @param $removedPermission
    */
   private function removePermissionFromMenuItem($menuItemName, $removedPermission) {
-    $reportsNavId = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', $menuItemName, 'id', 'name');
+    $menuItemID = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_Navigation', $menuItemName, 'id', 'name');
 
-    if ($reportsNavId) {
+    if ($menuItemID) {
       $navigation = new CRM_Core_DAO_Navigation();
-      $navigation->id = $reportsNavId;
+      $navigation->id = $menuItemID;
       $navigation->find(TRUE);
 
-      $permissions = explode(',', $navigation->permission);
+      $currentPermissions = explode(',', $navigation->permission);
 
-      foreach (array_keys($permissions, $removedPermission) as $key) {
-        unset($permissions[$key]);
+      foreach (array_keys($currentPermissions, $removedPermission) as $key) {
+        unset($currentPermissions[$key]);
       }
 
-      $navigation->permission = implode(',', $permissions);
+      $navigation->permission = implode(',', $currentPermissions);
 
-      if (count($permissions) < 2) {
+      if (count($currentPermissions) < 2) {
```

Comment:
please replace 2 with a variable with proper name that indicate what 2 means here

Reply:
I'm going to change the logic a bit and add a comment so it's a little more clear, but in this case, 2 just means 2.  As in more than 1, you need at least 2 permissions to use a an operator to combine them.